sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs

```diff
@@ -1996,6 +1998,461 @@ internal virtual void SetNameFieldsIfNull()
         }
         #endregion Set Access Control
 
+        #region Set Access Control Recursive
+        /// <summary>
+        /// The <see cref="SetAccessControlListRecursive"/> operation sets the
+        /// Access Control on a path and subpaths
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update" />.
+        /// </summary>
+        /// <param name="accessControlList">
+        /// The POSIX access control list for the file or directory.
+        /// </param>
+        /// <param name="batchSize">
+        /// Optional. If data set size exceeds batch size then operation will be split into multiple requests so that progress can be tracked.
+        /// Batch size should be between 1 and 2000. Default is 2000.
+        /// </param>
+        /// <param name="progressHandler">
+        /// Optional <see cref="Progress{ChangeAccessControlListPartialResult}"/> callback where caller can track progress of the operation
+        /// as well as collect paths that failed to change Access Control.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="ChangeAccessControlListResult"/> that contains summary stats of the operation.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual ChangeAccessControlListResult SetAccessControlListRecursive(
```

Comment:
What's the difference between Setting ACLs and Modifying ACLS?  Also, how would a user go about removing ACLs?

Reply:
"set" wipes out existing permissions and sets whatever has been specified. "modify" preserves existing permissions 
 
for example
existing permission is user::r-x
set user::rw- will result in user::rw-
modify user::rw- will result in user::rwx

Remove is out of scope here as we're hashing out details.

Reply:
@seanmcc-msft I added remove part. The semantics of acl string differ from what Set/Modify accept thus creating new data model for that.
The remove api accepts strings like:
mask
mask:
mask::
default:user
user:<oid>
default:user:<oid>

please note that removal of user , group and other is not allowed.